[identity providers] LDAP userSearchBase field must not be null

```
Caused by: java.lang.IllegalArgumentException: searchBase must not be null (an empty string is acceptable).
	at org.springframework.util.Assert.notNull(Assert.java:115)
	at org.springframework.security.ldap.search.FilterBasedLdapUserSearch.<init>(FilterBasedLdapUserSearch.java:80)
	at io.gravitee.am.identityprovider.ldap.authentication.spring.LdapAuthenticationProviderConfiguration.userSearch(LdapAuthenticationProviderConfiguration.java:69)
	at io.gravitee.am.identityprovider.ldap.authentication.spring.LdapAuthenticationProviderConfiguration$$EnhancerBySpringCGLIB$$31ffcfca.CGLIB$userSearch$4(<generated>)
	at io.gravitee.am.identityprovider.ldap.authentication.spring.LdapAuthenticationProviderConfiguration$$EnhancerBySpringCGLIB$$31ffcfca$$FastClassBySpringCGLIB$$e124556c.invoke(<generated>)
	at org.springframework.cglib.proxy.MethodProxy.invokeSuper(MethodProxy.java:228)
	at org.springframework.context.annotation.ConfigurationClassEnhancer$BeanMethodInterceptor.intercept(ConfigurationClassEnhancer.java:356)
	at io.gravitee.am.identityprovider.ldap.authentication.spring.LdapAuthenticationProviderConfiguration$$EnhancerBySpringCGLIB$$31ffcfca.userSearch(<generated>)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.springframework.beans.factory.support.SimpleInstantiationStrategy.instantiate(SimpleInstantiationStrategy.java:162)
	... 157 common frames omitted```

--- a/gravitee-am-gateway/gravitee-am-gateway-handlers/gravitee-am-gateway-handlers-oauth2/src/main/java/io/gravitee/am/gateway/handler/oauth2/provider/security/ClientBasedAuthenticationProvider.java
+++ b/gravitee-am-gateway/gravitee-am-gateway-handlers/gravitee-am-gateway-handlers-oauth2/src/main/java/io/gravitee/am/gateway/handler/oauth2/provider/security/ClientBasedAuthenticationProvider.java
@@ -31,6 +31,7 @@ import org.springframework.security.core.authority.AuthorityUtils;
 import org.springframework.security.oauth2.common.util.OAuth2Utils;
 import org.springframework.security.oauth2.provider.ClientDetails;
 import org.springframework.security.oauth2.provider.ClientDetailsService;
+import org.springframework.util.StringUtils;
 
 import java.util.Iterator;
 import java.util.Map;
@@ -77,13 +78,17 @@ public class ClientBasedAuthenticationProvider implements AuthenticationProvider
                         io.gravitee.am.identityprovider.api.AuthenticationProvider authenticationProvider =
                                 identityProviderManager.get(provider);
 
+                        if (authenticationProvider == null) {
+                            throw new BadCredentialsException("Unable to load authentication provider " + provider + ", an error occurred during the initialization stage");
+                        }
+
                         try {
                             user = authenticationProvider.loadUserByUsername(provAuthentication);
                             // set user identity provider source
                             details.put(RepositoryProviderUtils.SOURCE, provider);
                         } catch (Exception ex) {
-                            logger.info("Unable to authenticate user {} with provider {}",
-                                    authentication.getName(), provider);
+                            logger.info("Unable to authenticate user {} with provider {}", authentication.getName(), provider, ex);
+                            throw new BadCredentialsException(ex.getMessage(), ex);
                         }
                     }
 
@@ -91,14 +96,14 @@ public class ClientBasedAuthenticationProvider implements AuthenticationProvider
                         return new UsernamePasswordAuthenticationToken(user, provAuthentication.getCredentials(),
                                 AuthorityUtils.NO_AUTHORITIES);
                     }
-
-                    throw new BadCredentialsException("Bad credentials");
+                    throw new BadCredentialsException("No user found for providers " + StringUtils.collectionToDelimitedString(identities, ","));
                 }
             } catch (Exception ex) {
+                throw new BadCredentialsException(ex.getMessage(), ex);
             }
         }
 
-        throw new BadCredentialsException(authentication.getName());
+        throw new BadCredentialsException("No client found for authentication " + authentication.getName());
     }
 
     @Override

--- a/gravitee-am-identityprovider/gravitee-am-identityprovider-ldap/src/main/java/io/gravitee/am/identityprovider/ldap/LdapIdentityProviderConfiguration.java
+++ b/gravitee-am-identityprovider/gravitee-am-identityprovider-ldap/src/main/java/io/gravitee/am/identityprovider/ldap/LdapIdentityProviderConfiguration.java
@@ -31,7 +31,7 @@ public class LdapIdentityProviderConfiguration implements IdentityProviderConfig
 
     private String contextSourcePassword;
 
-    private String userSearchBase;
+    private String userSearchBase = "";
 
     private String userSearchFilter;
 

